Normcore 2.1 Documentation Updates

nicksavarese opened the pull request:
- Integrate interim copy edits and documentation updates from Max + copywriter

maxweisel committed:
Create test sidebar layout.

maxweisel committed:
Add test page.

maxweisel committed:
First-pass on all docs pages. Super super rough.

maxweisel committed:
Add first-pass at generating markdown files. Update the reference template to show the new information.

maxweisel committed:
Fix formatting on reference pages.

maxweisel committed:
Updated reference pages.

maxweisel committed:
Fix bug in handling < and > in xml docs.

maxweisel committed:
remove test page.

maxweisel committed:
Small tweaks + TODOs

maxweisel committed:
Add platform common questions.

maxweisel committed:
Pass on Realtime to hand off to copywriter.

nicksavarese committed:
Update feature/2.1-docs branch with edits from copywriter

- Update copy with edits from copywriter as of 7/22/2021
- Update all bullets/numbering formatting to be consistent (using '*'' for bullets)
- Standardized all TODOS into // TODO format

nicksavarese committed:
Update transport.md

- Fix typo in ## Why WebRTC is the best transport option

nicksavarese committed:
Fix link to Normcore Private page

maxweisel commented:
## **ARCHITECTURE:**

### CLIENT

- [ ] Add a diagram to illustrate the Realtime API / Room API / Transport separation

- [ ] What happened to that diagram with the internet cloud on it? That would be good here or in an Overview page

### TRANSPORT

- [ ] It looks like this [webrtcbyalex.com](http://webrtcbyalex.com/) domain is expired - can we use another source?

## **NORMCORE PRIVATE**

### AUTHORITATIVE SERVERS

- [ ] Maybe we make a graphic showing the architecture for this?

### **PRICING**

PRICING

- [ ] Insert graphic for pricing tiers

## REALTIME API

### Readme

Synchronizing a Game Object

- [ ] Link to RealtimeTransform or ownership document that explains the difference

- [ ] Record a short video of this

Instantiating a Prefab

- [ ]  I do think a nice graphic at the top of this section will really help the copy on this page make more sense.

- [ ] Put architecture image here that shows the relationship. Maybe link to the Architecture page too?

- [ ] Make sure the Client Architecture doc describes the Realtime/Room relationship

Synchronizing Custom Data

- [ ] Pair this guide down and link out to the RealtimeComponent / RealtimeModel docs.

The goal is to reduce this as much as possible so it feels simple + easy.

## **ROOM**

### Collections

- [ ] Link to the reference page for the collection

### Common Questions

- [ ] Create RealtimeCallback example

### Datastore

- [ ] This will probably be clearer if the caching layer is explained.

### Readme

- [ ] Graphic of Room + Datastore + Media API + RealtimeModel + Transport all the way down so folks can get an idea of how it all fits together

### Room

- [ ] Find the other doc where we say don't use RPC messages and link to that here (or maybe it was copy written in google docs?)

maxweisel committed:
Max copy tweaks

maxweisel committed:
Bring over Nick's changes.

maxweisel committed:
Updates to Transport. Still working.

nicksavarese committed:
Update copy and formatting to fix issues from markdown transfer

- More copy and formatting fixes from Google Doc -> Markdown transfer

nicksavarese committed:
More formatting fixes

nicksavarese committed:
Update google-cloud-installation.md

- Fix `--zone ` trailing space

nicksavarese committed:
Update readme.md

- Fix missing period on line 24

nicksavarese committed:
Update common-questions.md and realtimetransform.md

- Fix typo on line 30 in realtime/common-questions.md
- Fix trailing space on line 17 in realtimetransform.md

maxweisel committed:
Finish edits to Transport doc.

maxweisel committed:
Remove unnecessary line breaks.

maxweisel committed:
Make all headings in Transport the same width to keep the sidebar from being polluted.

maxweisel committed:
Edits to the pricing docs.

maxweisel committed:
Update sidebar to have Overview links instead of relying on the section title. Update Realtime overview doc.

maxweisel committed:
Copy edits to Realtime, RealtimeComponent, and RealtimeView.

maxweisel committed:
Updates to RealtimeTransform page.

maxweisel committed:
More RealtimeTransform edits.

maxweisel committed:
Synchronizing custom data edits.

maxweisel committed:
Tweaks to networked physics.

maxweisel committed:
Edits to Realtime common questions

maxweisel committed:
Updated copy for the avatars page.

maxweisel committed:
Copy changes to the voice chat section.

maxweisel committed:
Edits to the Room + Datastore API section.

maxweisel committed:
Final copy edits before we start on assets and final copy review.

maxweisel committed:
Implement timer + synchronized playback recipe.

maxweisel committed:
Drop docs on deprecated RealtimeView setting.

maxweisel committed:
Still working on addressables example.

maxweisel committed:
Finish Using Addressables code recipe. Just need to film assets.

maxweisel committed:
Finish up addressables recipe.

maxweisel committed:
Update reference docs related to realtime prefab delegates.

maxweisel committed:
Update all reference markdown pages.

maxweisel committed:
Update link to IRealtimePrefabLoadDelegate.

maxweisel committed:
First pass on rpc events recipe.

maxweisel committed:
Add videos to rpc events guide. Do another copy pass.

maxweisel committed:
Add project template files.

maxweisel committed:
Final copy edits to RealtimeCallback guide.

maxweisel committed:
Move pricing page.

nicksavarese committed:
Add Realtime Transform example video, update Realtime UI assets (#6)

* Add Realtime Transform example video, update Realtime UI assets

maxweisel committed:
Add Dave's graphics.

maxweisel committed:
Bring back diagram to synchronizing custom data guide.

maxweisel committed:
Update architecture diagram.

maxweisel committed:
Update getting started guide.

thewaterbear committed:
Added graphic with link to forum (#7)

maxweisel committed:
Remove old reference pages.

maxweisel committed:
Add edits from copywriter.

maxweisel committed:
Add remaining edits from copywriter.

maxweisel committed:
Fix W A S D -> WASD. Add Edit on GitHub links.

maxweisel committed:
Add forum links to the landing page. Tweak essentials feature list.
